model/src/pyrenew/process/periodiceffect.py

```diff
@@ -0,0 +1,161 @@
+# numpydoc ignore=GL08
+
+from typing import NamedTuple
+
+import jax.numpy as jnp
+import pyrenew.arrayutils as au
+from pyrenew.metaclass import RandomVariable, _assert_sample_and_rtype
+
+
+class PeriodicEffectSample(NamedTuple):
+    """
+    A container for holding the output from
+    `process.PeriodicEffect.sample()`.
+
+    Attributes
+    ----------
+    value: jnp.ndarray
+        The sampled value.
+    """
+
+    value: jnp.ndarray
+
+    def __repr__(self):
+        return f"PeriodicEffectSample(value={self.value})"
+
+
+class PeriodicEffect(RandomVariable):
+    """
+    Periodic effect with repeating values from a prior.
+    """
+
+    def __init__(
+        self,
+        data_starts: int,
+        period_size: int,
+        prior: RandomVariable,
+    ):
+        """
+        Default constructor for PeriodicEffect class.
+
+        Parameters
+        ----------
+        data_starts : int
+            Relative point at which data starts, must be between 0 and
+            period_size - 1.
+        period_size : int
+            Size of the period.
+        prior : RandomVariable
```

Comment:
suggest something more descriptive like `broadcast_values`

Reply:
Fixed in 1d1ab6b. I'm using `quantity_to_broadcast`, which is easy to replace, if you want to.